apps/btshell/src/cmd.c

```diff
@@ -99,12 +99,11 @@ parse_dev_addr(const char *prefix, const struct kv_pair *addr_types,
     char name[32];
     int rc;
 
-    /* XXX string operations below are not quite safe, but do we care? */
-
+    /* Valid prefixes are peer_, own_, or null. Let's copy up to 5 characters */
     if (!prefix) {
         name[0] = '\0';
     } else {
-        strcpy(name, prefix);
+        strncpy(name, prefix, 5);
```

Comment:
strncpy will not null terminate name if prefix is longer than 5 characters

Reply:
so maybe just return error when prefix is longer than 5 and avoid copying at all? longest viable prefix is "peer_"

Reply:
ENOENT for example

Reply:
I do not know if we won't use other prefixes in future. if you want to make it really fool-proof then better use strlcat (I think we support it) everywhere and handle errors.

Reply:
ok, switched to strlcat/cpy